pkg/argocd/git.go

```diff
@@ -58,11 +60,72 @@ func TemplateCommitMessage(tpl *template.Template, appName string, changeList []
 	return cmBuf.String()
 }
 
+// TemplateBranchName parses a string to a template, and returns a
+// branch name from that new template. If a branch name can not be
+// rendered, it returns an empty value.
+func TemplateBranchName(branchName string, changeList []ChangeEntry) string {
+	var cmBuf bytes.Buffer
+
+	tpl, err1 := template.New("branchName").Parse(branchName)
+
+	if err1 != nil {
+		log.Errorf("could not create template for Git branch name: %v", err1)
+		return ""
+	}
+
+	type imageChange struct {
+		Name   string
+		Alias  string
+		OldTag string
+		NewTag string
+	}
+
+	type branchNameTemplate struct {
+		Images []imageChange
+		SHA1   string
+	}
+
+	// Let's add a unique hash to the template
+	hasher := sha1.New()
```

Comment:
Can you please use `sha256` instead of `sha1`? I know the implications for this particular use-case are marginal, but SHA1 is going to go away rather soon and should not be used anymore.